CodeAssignment_MVVM_Combine/Sources/Views/Cells/HomeViewController/RepositoryCollectionViewCell.swift

```diff
@@ -0,0 +1,276 @@
+        view.clipsToBounds = true
+        view.backgroundColor = .systemPink
+        return view
+    }()
+    
+    private lazy var languageNameLabel: UILabel = {
+        let label = UILabel()
+        label.font = .systemFont(ofSize: 16, weight: .regular)
+        label.textColor = .systemGray
+        return label
+    }()
+    
+    // コードベースなので、ここで必須のfuncを実装する
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setupUI()
+    }
+    
+    // nibファイルを用いて実装していないことをfatalErrorを用いて明確にする
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    /** Cellレイアウトが確定されたタイミングで呼び出されるので、CellのSubViewのCornerRadiusのような設定が可能
+     - lazy varのクロージャ内ではビューのフレームがまだ決定されていないため、frame.heightを使用しても正しい値を取得できない
+     - そのため、layoutSubViewsをオーバーライドして設定する
+    */
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        userAccessoryView.layer.cornerRadius = userAccessoryView.frame.height / 2
+        userImageView.layer.cornerRadius = userImageView.frame.height / 2
+        languageColorView.layer.cornerRadius = languageColorView.frame.height / 2
+    }
+}
+
+// MARK: - Function and Logics
+extension RepositoryCollectionViewCell {
+    /** Cellのデータの設定を行う
+     - 今後、RepositoriesForViewのModelのデータに基づいて、Cellのデータを確立させる
+    */
+    func configure(with model: Repositories.Repository) {
+        let defaultImage = UIImage(systemName: "person.circle")?.withTintColor(.systemMint, renderingMode: .alwaysOriginal)
+        
+        nameLabel.text = model.name
+        descriptionLabel.text = model.description
+        userNameLabel.text = model.owner.userName
+        starCountsLabel.text = formatNumberToString(model.stargazersCount ?? 0)
+        languageNameLabel.text = model.language
+        
+        if let urlString = model.owner.profileImageString,
+           let url = URL(string: urlString) {
+            // completedに入れないと、常にplaceHolderImageしか表示されない仕組み
+            // completion handlerを用いて、画像のロード中にエラーが発生した場合や、URLが無効な場合にデフォルトの画像を表示する仕組みである
+            userImageView.sd_setImage(with: url, placeholderImage: defaultImage) { [weak self] (image, error, cacheType, url) in
+                guard let self else { return }
+                if let error = error {
+                    // ロード中にエラーが発生する場合や、URLが無効な場合はdefaultの画像を表示
+                    print("error: \(error.localizedDescription)")
+                    self.userImageView.image = defaultImage
+                }
+                self.userImageView.image = image
+            }
+        } else {
+            userImageView.image = defaultImage
+        }
+    }
+    
+    /// 数字が１万を超えたら、1.~万みたいにString型としてformatする
+    private func formatNumberToString(_ number: Int) -> String {
+        if number >= 10000 {
+            let formattedNumber = Double(number) / 10000
+            return String(format: "%.1f万", formattedNumber)
+        } else {
+            return String(number)
+        }
```

Comment:
三項演算子使えばこんな感じに1文で書ける

```suggestion
        number >= 10000 ? String(format: "%.1f万", Double(number) / 10000) : String(number)
```

Reply:
プラスで、今回1万を超えたらformatするっていう基準を作ってfuncの中に `10000` っていう数字を入れていると思うんだけど、こういう時は↓みたいにConstで名前を定義してあげると、なんで10000なのか、何の10000なのかがわかりやすくなる。

```
private enum Const {
   /// 数値を短縮表示する基準
   static let numberFormatThreshold: Int = 10000
}
```
```suggestion
       number >= Const.numberFormatThreshold ? String(format: "%.1f万", Double(number) / Const.numberFormatThreshold) : String(number)
```

Reply:
[b6e1f6b]にてコード修正しました！ご確認、お願いします！

Reply:
イメージ的にはCellのファイルの中にConstを置く感じだったけど、まあこれでもいいか